sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ldevice/LDeviceAdapterTest.java

```diff
@@ -39,55 +39,52 @@
+    private IEDAdapter iedAdapter;
 
     @BeforeEach
     public void init() {
         SCL scd = SclTestMarshaller.getSCLFromFile("/ied-test-schema-conf/ied_unit_test.xml");
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        iAdapter = assertDoesNotThrow(() -> sclRootAdapter.getIEDAdapterByName("IED_NAME"));
+        iedAdapter = assertDoesNotThrow(() -> sclRootAdapter.getIEDAdapterByName("IED_NAME"));
     }
 
     @Test
-    void updateLDName_when_ldName_pass_33_characters_should_throw_exception() {
+    void updateLDName_when_ldName_less_than_64_characters_should_not_throw_exception() {
         // Given
-        LDeviceAdapter lDeviceAdapter = assertDoesNotThrow(()-> iAdapter.findLDeviceAdapterByLdInst("LD_INS1").get());
+        SCL std = SclTestMarshaller.getSCLFromFile("/std/std_sample.std");
+        TIED tied = std.getIED().getFirst();
+        TLDevice tlDevice = tied.getAccessPoint().getFirst().getServer().getLDevice().getFirst();
+        LDeviceAdapter lDeviceAdapter = new LDeviceAdapter(new IEDAdapter(new SclRootAdapter(std), tied), tlDevice);
+
+        //When
         lDeviceAdapter.updateLDName();
-        assertThat(lDeviceAdapter.getLdName()).isEqualTo("IED_NAMELD_INS1");
-        assertThat(lDeviceAdapter.getInst()).isEqualTo("LD_INS1");
-        String iedName = new Random().ints(97, 122)
-                .limit(27)
-                .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
-                .toString();
-        iAdapter.setIEDName(iedName);
-        // When Then
-        assertThatThrownBy(lDeviceAdapter::updateLDName)
-                .isInstanceOf(ScdException.class)
-                .hasMessageContaining("has more than 33 characters");
+
+        //When
+        assertThat(lDeviceAdapter.getLdName()).isEqualTo("IED4d4fe1a8cda64cf88a5ee4176a1a0eefLDSUIED");
+        assertThat(lDeviceAdapter.getInst()).isEqualTo("LDSUIED");
     }
 
     @Test
-    void updateLDName_when_ldName_less_than_33_characters_should_not_throw_exception() {
+    void updateLDName_when_ldName_pass_64_characters_should_throw_exception() {
         // Given
-        LDeviceAdapter lDeviceAdapter = assertDoesNotThrow(()-> iAdapter.findLDeviceAdapterByLdInst("LD_INS1").get());
-        lDeviceAdapter.updateLDName();
-        assertThat(lDeviceAdapter.getLdName()).isEqualTo("IED_NAMELD_INS1");
-        assertThat(lDeviceAdapter.getInst()).isEqualTo("LD_INS1");
-        String iedName = new Random().ints(97, 122)
-                .limit(26)
-                .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
-                .toString();
-        iAdapter.setIEDName(iedName);
+        SCL std = SclTestMarshaller.getSCLFromFile("/std/std_sample.std");
+        TIED tied = std.getIED().getFirst();
+        TLDevice tlDevice = tied.getAccessPoint().getFirst().getServer().getLDevice().getFirst();
+        tlDevice.setInst("LDSUIED***********************");
```

Comment:
This is not a valid pattern, it can be good test for the [pattern]( https://github.com/com-pas/compas-scl-xsd/blob/develop/src/main/resources/xsd/SCL2007B4/SCL_BaseSimpleTypes.xsd#L103) restriction, not for the [maxLength]( https://github.com/com-pas/compas-scl-xsd/blob/develop/src/main/resources/xsd/SCL2007B4/SCL_BaseSimpleTypes.xsd#L102) restriction.

Reply:
changed the pattern